modules/miner/update.go

```diff
@@ -28,11 +27,138 @@ func (m *Miner) addNewTxns(diff *modules.TransactionPoolDiff) {
 				totalFees = totalFees.Add(fee)
 			}
 		}
-		m.splitSets[m.setCounter] = &splitSet{
+		// We will check to see if this splitSet belongs in the block.
+		s := &splitSet{
 			size:         size,
 			averageFee:   totalFees.Div64(size),
 			transactions: newSet.Transactions,
 		}
+
+		elem := &mapElement{
+			set:   s,
+			id:    splitSetID(m.setCounter),
+			index: 0,
+		}
+
+		m.splitSets[splitSetID(m.setCounter)] = s
+		newElements = append(newElements, elem)
```

Comment:
I think it's bad design to add the splitSet to the miner state (`m.splitSets`) but then not add the newElements to the global state at the same time. You want all of your dependent state modifications to happen at the same time. Probably better to return an array of splitSets alongside the array of mapElements, and fill out `m.splitSets` in the same function that you add the map elements to the heaps.